tls/s2n_handshake_io.c

```diff
@@ -812,6 +813,29 @@ int s2n_conn_set_handshake_type(struct s2n_connection *conn)
             if (!s2n_decrypt_session_ticket(conn, &conn->client_ticket_to_decrypt)) {
                 return S2N_SUCCESS;
             }
+            s2n_extension_type_id ems_ext_id = 0;
+            POSIX_GUARD(s2n_extension_supported_iana_value_to_id(TLS_EXTENSION_EMS, &ems_ext_id));
+            /* TODO: https://github.com/aws/s2n-tls/issues/2990 */
+            if (conn->ems_negotiated && S2N_IN_TEST) {
+                /**
+                 *= https://tools.ietf.org/rfc/rfc7627#section-5.3
+                 *# If the original session used the "extended_master_secret"
+                 *# extension but the new ClientHello does not contain it, the server
+                 *# MUST abort the abbreviated handshake.
+                 **/
+                POSIX_ENSURE(S2N_CBIT_TEST(conn->extension_requests_received, ems_ext_id), S2N_ERR_MISSING_EXTENSION);
```

Comment:
Because we process the extensions before we decrypt the session ticket. So when we call if_missing, we don't actually know whether or not the session ticket is an EMS session ticket or not.